velox/functions/prestosql/GeometryFunctions.h

```diff
@@ -109,6 +113,28 @@ struct StAsBinaryFunction {
   }
 };
 
+template <typename T>
+struct StPointFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE Status call(
+      out_type<Geometry>& result,
+      const arg_type<double>& x,
+      const arg_type<double>& y) {
+    if (!std::isfinite(x) || !std::isfinite(y)) {
+      return Status::UserError(fmt::format(
+          "ST_Point requires finite coordinates, got x={} y={}", x, y));
+    }
+    geos::geom::GeometryFactory::Ptr factory =
+        geos::geom::GeometryFactory::create();
+    geos::geom::Point* point =
```

Comment:
just curious, does `factory` own the memory allocated for point? like, would the memory be destroyed along with `factory` or whether its just a heap allocated naked pointer

Reply:
In 3.10.2, it returns a raw pointer that needs to be explicitly destroyed.  In 3.13.1, it returns an owned `std::unique_ptr`.

Reply:
We are setting the version to 3.13.1. But for compatibility, we can probably do `auto point = std::unique_ptr...()`